tezos/interop_callback/src/callback.rs

```diff
@@ -146,47 +146,53 @@ caml!(ml_context_commit(parent_context_hash, block_hash, new_context_hash, time_
     let block_hash: Option<BlockHash> = to_hash(block_hash.into());
     let operation_hash: Option<OperationHash> = to_hash(operation_hash.into());
-    let key: ContextKey = List::from(key).convert_to();
+    let keyval: Tuple = keyval.into();
+    let key: ContextKey = List::from(keyval.get(0).unwrap()).convert_to();
+    let value: bool = keyval.get(1).unwrap().i32_val() > 0;
 
     let time_period: Tuple = time_period.into();
     let start_time: f64 = time_period.get(0).unwrap().f64_val();
     let end_time: f64 = time_period.get(1).unwrap().f64_val();
 
-    context_mem(context_hash, block_hash, operation_hash, key, start_time, end_time);
+    context_mem(context_hash, block_hash, operation_hash, key, value, start_time, end_time);
     return Value::unit();
 });
 
 // External callback function for dir_mem key from context
-caml!(ml_context_dir_mem(context_hash, block_hash, operation_hash, key, time_period) {
+caml!(ml_context_dir_mem(context_hash, block_hash, operation_hash, keyval, time_period) {
     let context_hash: Option<ContextHash> = to_hash(context_hash.into());
     let block_hash: Option<BlockHash> = to_hash(block_hash.into());
     let operation_hash: Option<OperationHash> = to_hash(operation_hash.into());
-    let key: ContextKey = List::from(key).convert_to();
+    let keyval: Tuple = keyval.into();
+    let key: ContextKey = List::from(keyval.get(0).unwrap()).convert_to();
+    let value: bool = keyval.get(1).unwrap().i32_val() > 0;
 
     let time_period: Tuple = time_period.into();
     let start_time: f64 = time_period.get(0).unwrap().f64_val();
     let end_time: f64 = time_period.get(1).unwrap().f64_val();
 
-    context_dir_mem(context_hash, block_hash, operation_hash, key, start_time, end_time);
+    context_dir_mem(context_hash, block_hash, operation_hash, key, value, start_time, end_time);
     return Value::unit();
 });
 
 // External callback function for raw_get key from context
-caml!(ml_context_raw_get(context_hash, block_hash, operation_hash, key, time_period) {
+caml!(ml_context_raw_get(context_hash, block_hash, operation_hash, keyval, time_period) {
     let context_hash: Option<ContextHash> = to_hash(context_hash.into());
     let block_hash: Option<BlockHash> = to_hash(block_hash.into());
     let operation_hash: Option<OperationHash> = to_hash(operation_hash.into());
-    let key: ContextKey = List::from(key).convert_to();
+    let keyval: Tuple = keyval.into();
+    let key: ContextKey = List::from(keyval.get(0).unwrap()).convert_to();
+    let value: ContextValue = OcamlBytes::from(keyval.get(1).unwrap()).convert_to();
 
     let time_period: Tuple = time_period.into();
     let start_time: f64 = time_period.get(0).unwrap().f64_val();
     let end_time: f64 = time_period.get(1).unwrap().f64_val();
 
-    context_raw_get(context_hash, block_hash, operation_hash, key, start_time, end_time);
+    context_raw_get(context_hash, block_hash, operation_hash, key, value, start_time, end_time);
```

Comment:
@fafk @jurajselep 
1. pls. do a refactor keyval also for context_raw_set like I mentioned in gitlab PR
2. add also value_as_json to get result like in set result
```
        value: Vec<u8>,
        value_as_json: Option<String>,
```

Reply:
I got stuck on missing `protocol_hash` when I tried to compute the JSON value for GET with `resolve_value_as_json()`

It does on `let protocol_hash = Option.unopt_exn (Ffi_errors.Ffi_error "protocol_hash not set!") protocol_hash in`

I dug around and it seems that it's set when a block is being applied, but not quite sure how to proceed here. 🤔

Reply:
As for 1.: I think you didn't actually mention it in the GitLab PR (or I didn't mange to find it), but I aligned the GET and SET function annotations to be coherent.